### PR TITLE
 pkp/pkp-lib#8406 Improve DublinCoreMetaPlugin and adapt GoogleScholarPlugin

### PR DESCRIPTION
**Pull request by bozana**
s. https://github.com/pkp/pkp-lib/issues/8406

### CODE REVIEW

File: plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php [hunk 1]
```diff
@@ -181,110 +190,129 @@ public function monographFileView($hookName, $args)
...
         $request = Application::get()->getRequest();
         $monograph = $args[1];
-        $publication = $monograph->getCurrentPublication();
         $publicationFormat = $args[2];
         $submissionFile = $args[3];
+        $requestArgs = $request->getRequestedArgs();
         $press = $request->getContext();
 
+        // Only add Google Scholar metadata tags to the canonical URL for the latest version
+        // See discussion: https://github.com/pkp/pkp-lib/issues/4870
```
Comment (asmecher) [inline plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php:199]: Looking at the discussion there, should this not also apply to OJS and OPS?

**Review by asmecher — CHANGES_REQUESTED** [review]
Thanks, @bozana! Comments on https://github.com/pkp/ojs/pull/3746/files also apply here.

File: plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php (hunk 1 above)
Comment (bozana) [reply plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php:199]: Yes, it is applied everywhere...

File: plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php [hunk 2]
```diff
@@ -188,20 +193,21 @@ public function monographView($hookName, $args)
...
         $submissionFile = $args[3];
-        $requestArgs = $request->getRequestedArgs();
-        $press = $request->getContext();
+
+        $publication = $monograph->getCurrentPublication();
 
         // Only add Google Scholar metadata tags to the canonical URL for the latest version
         // See discussion: https://github.com/pkp/pkp-lib/issues/4870
-        if (count($requestArgs) > 1 && $requestArgs[1] === 'version') {
+        if ($publicationFormat->getData('publicationId') != $publication->getId()) {
```
Comment (bozana) [inline plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php:204]: I realized that there will be no 'version' in URL when a file is viewed, so I am checking this way if it is a file i.e. publication format of a current version/publication.

**Pull request merged**